internal/core/software_renderer.rs

```diff
@@ -394,6 +394,31 @@ mod private_api {
         ) -> bool {
             false
         }
+
+        /// Draw a texture into the buffer at the specified position with the given size and
+        /// colorized if needed. Returns true if the operation was successful; false if it could not be
+        /// implemented and instead the software renderer needs to draw the texture
+        fn draw_texture(
+            &mut self,
+            _x: i16,
+            _y: i16,
+            _width: i16,
+            _height: i16,
+            _src: *const u8,
```

Comment:
Functions that take raw pointer should be marked `unsafe`
This should ideally take a slice instead, for safety.

Actually, we probably also don't want to expose things like dx and dy which is really low level and specific to our implementation which we might need to change. Better structures should be used in the API.